### ISSUE TITLE
Enhancement of Test Cases

### ISSUE THREAD

**Issue by wzshiming** [kubernetes-sigs/kwok#1062]
### What would you like to be added?

https://mentorship.lfx.linuxfoundation.org/project/1969ce6c-5468-4842-89a2-06c020bd2ad1

The following tests are currently under consideration:
- Unit Test
- E2E Test
- Edge Cases
- Migrate shell’s test to e2e

### Why is this needed?

KWOK is currently being used by a number of projects for testing and performance. It is crucial that KWOK itself behaves consistently.

ref cncf/mentoring#1227

**Comment by ShivamJha2436** [comment]
Hello @wzshiming,
I'm Shivam, a developer passionate about contributing to the Kubernetes ecosystem through LFX. The KWOK project's focus on scalable cluster management aligns perfectly with my interests. While I'm still developing my expertise in Golang and Kubernetes testing, I'm eager to learn and contribute to improving KWOK's test coverage through unit, E2E, and edge case testing. Participating in this project under the LFX mentorship program would be an invaluable opportunity to gain practical experience alongside seasoned developers and ensure KWOK's continued reliability.
with regards,
Shivam Kumar

**Comment by wzshiming** [comment]
Thank you, but this program is [pending approval](https://github.com/cncf/mentoring/pull/1227#issuecomment-2099007423), if this is approved, Monday 13th May to start mentee applications.

https://github.com/cncf/mentoring/blob/main/programs/lfx-mentorship/2024/02-Jun-Aug/README.md

**Comment by ShivamJha2436** [comment]
Ok, thanks for letting me know.

**Comment by wzshiming** [comment]
The application is already open
https://mentorship.lfx.linuxfoundation.org/project/1969ce6c-5468-4842-89a2-06c020bd2ad1

**Comment by 2008sahil** [comment]
Hello @wzshiming ,

I hope you're doing well. My name is Sahil Gupta, and I'm a BTech undergraduate student with a strong interest in Kubernetes and software testing. I recently came across the CNCF - KWOK project and I'm excited about the opportunity to contribute.

Having worked extensively with Kubernetes clusters on Microsoft Azure in a recent project, I've gained practical experience in deploying and managing containerized applications at scale. Additionally, I've been actively learning Golang, recognizing its importance within the Kubernetes ecosystem.

The idea of enhancing test coverage for KWOK, especially through unit tests, end-to-end tests, and addressing edge cases, resonates with me. I believe that by improving the reliability and robustness of KWOK, we can positively impact various projects relying on its capabilities for testing and performance evaluation.

I'm particularly interested in writing tests for KWOK and would love to contribute in this aspect of the project. Could you please provide guidance on how I can get started and make meaningful contributions?

Thank you for considering my interest in participating in the CNCF - KWOK project. I'm looking forward to collaborating with you and the team to drive this project forward.

Best regards,
Sahil Gupta

**Comment by joeyyy09** [comment]
Hey everyone!
I am Harshith Mente, an undergraduate student at IIITM in India, with a strong passion for constructing and automating intricate systems. Recently, I came across KWOK while exploring CNCF projects for LFX, and the idea of swiftly creating massive simulated Kubernetes clusters truly captivated me.

Background and Expertise:
With a solid foundation in Kubernetes (k8s) and Golang, I am confident in my ability to provide valuable insights to the KWOK project. I am excited to delve deeper into KWOK's codebase and functionalities to begin making meaningful contributions.

Understanding of KWOK:
Purpose: KWOK (Kubernetes WithOut Kubelet) is akin to a magical tool for swiftly setting up large clusters of simulated computer nodes.
Tools:
kwok: Facilitates the creation of fake nodes and pods for experimentation.
kwokctl: Acts as a helpful assistant in managing these simulated clusters efficiently.
Standout Features of KWOK:
Differentiation from kubemark:

kubemark: Mimics a kubelet but consumes substantial memory during node simulation.
KWOK: Focuses on emulating nodes only, utilizing minimal memory even with numerous nodes.
Differentiation from kind:

kind: Constructs a genuine Kubernetes cluster within Docker.
KWOK: Serves as an alternative to kind when actual pod execution is unnecessary.
Contribution Goals:
I am keen on enhancing the test cases within KWOK and would greatly appreciate any guidance on identifying suitable introductory issues. Your assistance in this regard would be immensely valuable as I strive to comprehend the project thoroughly and contribute effectively.

Personal Links:
GitHub: github.com/joeyyy09
LinkedIn: LinkedIn Profile
Portfolio: Harshith Mente Portfolio

I am eager to collaborate with the KWOK community, learn, and make a positive impact. Thank you for your time and consideration!

**Comment by wzshiming** [comment]
The application is already open, If you want to be a mentee of LFX, apply here
https://mentorship.lfx.linuxfoundation.org/project/1969ce6c-5468-4842-89a2-06c020bd2ad1

**Comment by Psingle20** [comment]
@wzshiming could you send me the invite link to slack for kowk 
would like to discuss few things with about migrating shells test to e2e.
You can mail the invite link at ingleprachit101@gmail.com

**Comment by wzshiming** [comment]
> @wzshiming could you send me the invite link to slack for kowk would like to discuss few things with about migrating shells test to e2e. You can mail the invite link at [ingleprachit101@gmail.com](mailto:ingleprachit101@gmail.com)

https://kwok.sigs.k8s.io/#getting-involved
Here are the steps

**Comment by yashwanth1208** [comment]
Hello @wzshiming 

I'm Yashwanth Sai Sollu, currently pursuing my Bachelor of Technology in Computer Science and Engineering at VNR Vignana Jyothi Institute of Engineering and Technology in Hyderabad, India. With a passion for software development and problem-solving, I've embarked on a journey to explore the intricacies of technology and its applications in solving real-world challenges.

Background and Expertise:
Equipped with a diverse skill set, I've honed my abilities in various programming languages such as Python, Go Lang, Java, and JavaScript. My experiences range from participating in competitive coding contests like Google Code Jam to collaborating on innovative projects like XeroDrop and BlockKicks. These experiences have not only strengthened my technical prowess but also instilled in me a drive to create impactful solutions that benefit society.

Understanding of Technology:
My journey in technology has led me to delve into a myriad of domains, from machine learning and blockchain to full-stack development. I've embraced the complexity of these fields, leveraging tools and frameworks like ReactJS, Flutter, and TensorFlow to build scalable and secure applications. I'm particularly drawn to the intersection of technology and social good, as evidenced by my project BlockKicks, which aims to combat counterfeit products using blockchain technology.

Contribution Goals:
As I embark on this mentorship journey, I am eager to immerse myself in the KWOK project and contribute to its growth and success. I am keen on enhancing the test cases within KWOK and leveraging my skills to identify and address challenges effectively. Your guidance and mentorship will be invaluable as I navigate through the project and strive to make meaningful contributions to the KWOK community.

Personal Links:
GitHub: https://github.com/yashwanth1208
LinkedIn: https://www.linkedin.com/in/yashwanth-sai-sollu-009a50201/

Additionally, I have applied as a mentee on LFX, and I am excited about the opportunity to further develop my skills and contribute to open-source projects through mentorship programs.

I am excited to collaborate with the KWOK community, learn, and make a positive impact. Thank you for considering my application and for the opportunity to be a part of this enriching mentorship program!

Best regards,
Yashwanth Sai Sollu
